File: HelloCSharp/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace HelloCSharp
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        public static string myText;
        private void cmdHello_Click(object sender, EventArgs e)
        {
            //if (txtName.Text == "")
            //    {
            //    ErrorProvider1.SetError(txtName, "Name Required");
            //    }
            //else
            //{
            //    ErrorProvider1.SetError(txtName, "");
            //    MessageBox.Show("Hello " + txtName.Text + "\nThe time is: " + DateTime.Now.ToString("h:mm:ss"));
            //    this.Text = "Hello " + txtName.Text;
            //}

            Form2 f2 = new Form2();
            f2.ShowDialog();
            txtName.Text = myText;


        }
    }
}

File: HelloVBNet/Form1.vb
Public Class Form1
    Private Sub cmdHello_Click(sender As Object, e As EventArgs) Handles cmdHello.Click
        If txtName.Text = "" Then
            ErrorProvider1.SetError(txtName, "Name Required")
            Exit Sub
        Else
            ErrorProvider1.SetError(txtName, "")
        End If

        MessageBox.Show("Hello " & txtName.Text & vbNewLine & "The time is: " & TimeOfDay)
        Me.Text = "Hello " & txtName.Text
    End Sub
End Class
